include/lsst/afw/geom/Endpoint.h

```diff
@@ -0,0 +1,303 @@
+// -*- lsst-c++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2008-2017 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_AFW_GEOM_ENDPOINT_H
+#define LSST_AFW_GEOM_ENDPOINT_H
+
+#include <memory>
+#include <vector>
+
+#include "astshim.h"
+#include "ndarray.h"
+
+#include "lsst/afw/geom/Point.h"
+#include "lsst/afw/geom/SpherePoint.h"
+
+namespace lsst {
+namespace afw {
+namespace geom {
+
+
+/**
+Virtual base class for endpoints, which are helper classes for Transform
+
+Endpoints transform points and lists of points from LSST-specific data types,
+such as Point2D and SpherePoint, to a form accepted by ast::Mapping.tran.
+Each type of endpoint is used for a particular LSST data type, for example:
+- Point2Endpoint is used for Point2D data
+- Point3Endpoint is used for Point3D data
+- SpherePointEndpoint for SpherePoint data
+- GenericEndpoint is used when no other form will do; its LSST data type
+  is identical to the type used for ast::Mapping.tran.
```

Comment:
I prefer my wording. I consider it an explanation of the design and a suggestion of a place to look for more information.